os/arch/arm/src/amebasmart/amebasmart_idle.c

```diff
@@ -23,14 +23,160 @@
  ****************************************************************************/
 
 #include <tinyara/config.h>
-
 #include <tinyara/arch.h>
-#include "up_internal.h"
+#include <tinyara/irq.h>
+
+#ifdef CONFIG_PM
+#include <tinyara/pm/pm.h>
+#include "amebasmart_config.h"
+#include "arch_timer.h"
+#include "ameba_soc.h"
+#include "osdep_service.h"
+#endif
+
+#ifdef CONFIG_SMP
+#include "gic.h"
+#endif
 
 /****************************************************************************
- * Public Functions
+ * Name: up_idlepm
+ *
+ * Description:
+ *   Perform IDLE state power management.
+ *
  ****************************************************************************/
 
+#ifdef CONFIG_PM
+static enum pm_state_e oldstate = PM_NORMAL;
+static void up_idlepm(void)
+{
+	uint32_t xModifiableIdleTime = 0;
+	enum pm_state_e newstate;
+	irqstate_t flags;
+	int ret;
+
+	/* Decide, which power saving level can be obtained */
+	newstate = pm_checkstate(PM_IDLE_DOMAIN);
+
+	/* Check for state changes */
+	if (newstate != oldstate)
+	{
+		//TODO: Critical section code needed for SMP case?
```

Comment:
You can remove any such additional comments that I had added initially while pushing the first skeleton.
If we are pushing PM code without completely handling SMP then we can let this TODO be, but other comment lines below this should still be resolved and removed.